tests/test_reset.py

```diff
@@ -0,0 +1,154 @@
+"""
+This file was adapted from a setup file given in 
+the https://github.com/microsoft/AirSim-Drone-Racing-Lab repositiry
+It simply loads a level and starts a race against a baseline drone
+to validate that everything has been set up properly.
+"""
+
+import airsimdroneracinglab
+import threading
+import time
+
+
+class ReproduceResetRaceCondition:
+    """
+    Contains vaious functions for interacting with the ADRL unreal environment
+    """
+
+    def __init__(self, drone_name="drone_1"):
+        """
+        Initializes class variables
+
+        Args:
+        -----
+            drone_name (string) = Name of user controlled drone (defaults to 'drone_1')
+        """
+        self.airsim_client = airsimdroneracinglab.MultirotorClient()
+        self.airsim_client_2 = airsimdroneracinglab.MultirotorClient()
+        self.airsim_client_3 = airsimdroneracinglab.MultirotorClient()
+        self.drone_name = drone_name
+        self.is_thread_active = False
+        self.thread_reset = threading.Thread(
+            target=self.repeat_timer, args=(self.reset, 0.05)
+        )
+        self.thread_reset_race = threading.Thread(
+            target=self.repeat_timer, args=(self.reset_race, 0.03)
+        )
+        self.thread_reset_and_reset_race = threading.Thread(
+            target=self.repeat_timer, args=(self.reset_and_reset_race, 0.09)
+        )
+        self.is_thread_active = False
+
+    def repeat_timer(self, callback, period):
+        """
+        Simple sleep timer
+        
+        Args:
+        -----
+            callback (function) = Function to call
+            period (float) = Repeat interval in seconds
+        """
```

Comment:
Same deal here as before;
```suggestion
    def repeat_timer(self, callback, period: float):
        """
        Simple sleep timer
        
        Parameters
        ------------
            callback = Function to call
            period: float = Repeat interval in seconds
        """
```

Reply:
Passing functions into other functions is usually a no-no, but I'll trust you here